src/models/converters/transactions/mod.rs

```diff
@@ -19,58 +21,17 @@ use crate::providers::info::{InfoProvider, SafeInfo, TokenInfo, TokenType};
 use crate::utils::TRANSFER_METHOD;
 use rocket::futures::future::OptionFuture;
 
-impl MultisigTransaction {
-    fn confirmation_count(&self) -> u64 {
-        match &self.confirmations {
-            Some(confirmations) => confirmations.len() as u64,
-            None => 0,
-        }
-    }
-
-    fn confirmation_required(&self, threshold: u64) -> u64 {
-        self.confirmations_required.unwrap_or(threshold)
-    }
-
-    fn missing_signers(&self, owners: &Vec<String>) -> Vec<String> {
-        self.confirmations.as_ref().map_or_else(
-            || owners.to_owned(),
-            |confirmations| {
-                owners
-                    .iter()
-                    .filter_map(|owner| {
-                        if !confirmations.iter().any(|c| &c.owner == owner) {
-                            Some(owner.to_owned())
-                        } else {
-                            None
-                        }
-                    })
-                    .collect()
-            },
-        )
-    }
-
-    fn map_status(&self, safe_info: &SafeInfo) -> TransactionStatus {
-        if self.is_executed {
-            if self.is_successful.unwrap_or(false) {
-                TransactionStatus::Success
-            } else {
-                TransactionStatus::Failed
-            }
-        } else if safe_info.nonce > self.nonce {
-            TransactionStatus::Cancelled
-        } else if self.confirmation_count() < self.confirmation_required(safe_info.threshold) {
-            TransactionStatus::AwaitingConfirmations
-        } else {
-            TransactionStatus::AwaitingExecution
-        }
-    }
-
-    async fn transaction_info(&self, info_provider: &impl InfoProvider) -> TransactionInfo {
+impl SafeTransaction {
+    async fn transaction_info(
+        &self,
+        info_provider: &impl InfoProvider,
+        is_cancellation: bool,
+    ) -> TransactionInfo {
```

Comment:
I realise that this is not part of your changes; but I am a it surprise we don't wrap the result value of this function in `ApiResult` 😁